Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python
+# ===================================
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See license.txt for license information.
+# ===================================
+
+import socket
+import os
+import sys
+import imp
+import hashlib
+import codecs
+import base64
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+LG = nxDSCLog.DSCLog
+
+global show_mof
+show_mof = False
+CONFIG_PATH = '/etc/opt/microsoft/omsagent/conf/'
+SERVER_ADDRESS = '/var/opt/microsoft/omsagent/run/npmdagent.sock'
+DEST_FILE_NAME = 'npmd_agent_config.conf'
+
+#   [key] string ConfigType;
+#   [write] string ConfigID;
+#   [write] string Contents;
+#   [write,ValueMap{"Present", "Absent"},Values{"Present", "Absent"}] string Ensure;
+#   [write] string ContentChecksum;
+
+def init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if ConfigType is not None and ConfigType != '':
+        ConfigType = ConfigType.encode('ascii', 'ignore')
+    else:
+        ConfigType = 'UpdateNPMAgentConfig'
+    if ConfigID is not None:
+        ConfigID = ConfigID.encode('ascii', 'ignore')
+    else:
+        ConfigID = ''
+    if Contents is not None:
+        Contents = base64.b64decode(Contents)#Contents.encode('ascii', 'ignore')
+    else:
+        Contents = ''
+    if Ensure is not None and Ensure != '':
+        Ensure = Ensure.encode('ascii', 'ignore')
+    else:
+        Ensure = 'Present'
+    if ContentChecksum is not None:
+        ContentChecksum = ContentChecksum.encode('ascii', 'ignore')
+    else:
+        ContentChecksum = ''
+    return ConfigType, ConfigID, Contents, Ensure, ContentChecksum
+
+
+def Set_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    (ConfigType, ConfigID, Contents, Ensure, ContentChecksum) = init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    retval = Set(ConfigType, ConfigID, Contents, Ensure, ContentChecksum)
+    return retval
+
+
+def Test_Marshall(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
```

Comment:
@varastog - I would recommend that you add tests with this commit, The tests will help you from making sure that your resource works on all version of python. It will also in a way force you to come up with your config MOF, which you can use to work your way up for AMS changes. Also in future, when you will reference this commit for similar changes, you will have one complete commit for your self.